ngi_pipeline/engines/qc_ngi/launchers.py

```diff
@@ -0,0 +1,144 @@
+"""Run the qc pipeline on fastq samples; currently consists of FastQC and
+fastq_screen."""
+
+import os
+import subprocess
+
+from ngi_pipeline.engines.qc_ngi.workflows import return_cl_for_workflow
+from ngi_pipeline.log.loggers import log_process_non_blocking, minimal_logger
+from ngi_pipeline.utils.classes import with_ngi_config
+from ngi_pipeline.utils.filesystem import rotate_file, safe_makedir
+from ngi_pipeline.utils.parsers import find_fastq_read_pairs
+
+LOG = minimal_logger(__name__)
+
+
+@with_ngi_config
+def analyze(project, sample, config=None, config_file_path=None):
+    """The main entry point for the qc pipeline."""
+    LOG.info("Processing project/sample {}/{}".format(project, sample))
+
+    # Two paths diverged in a yellow wood
+    project_analysis_path = os.path.join(project.base_path,
+                                         "ANALYSIS",
+                                         project.project_id,
+                                         "qc_ngi")
+    # and sorry I could not travel both
+    sample_analysis_path = os.path.join(project_analysis_path, sample.name)
+    # and be one traveler, long I stood
+    log_dir_path = os.path.join(project_analysis_path, "logs")
+    # and looked down one as far as I could
+    safe_makedir(sample_analysis_path)
+    # To where it bent in the undergrowth
+    safe_makedir(log_dir_path)
+    # I need to go to sleep
+
+    fastq_files_to_process = []
+    # I suppose I -should- have quoted the other one
+    src_fastq_base = os.path.join(project.base_path, "DATA",
+                                  project.project_id, sample.name)
+    # Whose woods these are I think I know
+    for libprep in sample:
+        # His house is in the village though
+        for seqrun in libprep:
+            # He will not see mt stopping here
+            for fastq_file in seqrun:
+                # To watch
+                path_to_src_fastq = os.path.join(src_fastq_base,
+                                                 libprep.name,
+                                                 seqrun.name,
+                                                 fastq_file)
+                # his woods
+                fastq_files_to_process.append(path_to_src_fastqc)
+    # fill up
+    paired_fastq_files = find_fastq_read_pairs(fastq_files_to_process).values()
```

Comment:
hope this comment will not add too much complexity.... as we want in the future to run this on all samples we produced (not only WHG) we need to be able to process also single ended reads